### ISSUE TITLE
After upgrade to 1.07 Pydebrid fails completely

### ISSUE THREAD

**Issue by FrankenLab** [davidderus/pydebrid#4]
Command was: pydebrid --stdout -u http://blah-blah-blah

Traceback (most recent call last):
  File "/usr/local/bin/pydebrid", line 9, in <module>
    load_entry_point('pydebrid==1.0.7', 'console_scripts', 'pydebrid')()
  File "/usr/local/lib/python2.7/site-packages/pkg_resources/**init**.py", line 552, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/local/lib/python2.7/site-packages/pkg_resources/**init**.py", line 2672, in load_entry_point
    return ep.load()
  File "/usr/local/lib/python2.7/site-packages/pkg_resources/**init**.py", line 2345, in load
    return self.resolve()
  File "/usr/local/lib/python2.7/site-packages/pkg_resources/**init**.py", line 2351, in resolve
    module = **import**(self.module_name, fromlist=['**name**'], level=0)
  File "/usr/local/lib/python2.7/site-packages/pydebrid/main.py", line 12, in <module>
    from core import Alldebrid, PydConfig, URI, Downloader, Subtitle
  File "/usr/local/lib/python2.7/site-packages/pydebrid/core/**init**.py", line 5, in <module>
    from subtitle import Subtitle
  File "/usr/local/lib/python2.7/site-packages/pydebrid/core/subtitle.py", line 4, in <module>
    import subliminal
  File "/usr/local/lib/python2.7/site-packages/subliminal/**init**.py", line 10, in <module>
    from .api import (ProviderPool, check_video, provider_manager, download_best_subtitles, download_subtitles,
  File "/usr/local/lib/python2.7/site-packages/subliminal/api.py", line 14, in <module>
    from .subtitle import compute_score, get_subtitle_path
  File "/usr/local/lib/python2.7/site-packages/subliminal/subtitle.py", line 7, in <module>
    from guessit.matchtree import MatchTree
ImportError: No module named matchtree

**Comment by FrankenLab** [comment]
I think I found a clue, but the solution evades me at the moment.  Its a version mismatch between Subliminal and Guessit. Subliminal requires Guessit 1.x ? but Guessit has been upgraded to 2.0.4.

**Comment by FrankenLab** [comment]
Doing both of the below commands seems to have fixed the problem and allowed pydebrid to work again.

pip install 'guessit==1.0.2' --force-reinstall
pip install keyrings.alt

**Comment by davidderus** [comment]
The subliminal version used by pydebrid (_v1.0.1_) does not set an upper limit for the guessit dependancy but is incompatible with guessit >= 2.0.0.

So when installing the new pydebrid update, pip installs the last available guessit version (_>=2.0.0_), thus creating your error.

Forcing the guessit version system-wide by doing `pip install 'guessit==1.0.2' --force-reinstall` is a solution, but a more permanent one will be to update the subliminal version in pydebrid requirements.
It just need a few checks in order to avoid breaking the whole thing.

Thus we will have both new subliminal features/bugfixes and a patch for this issue, as they have set an upper limit starting 1.1.

I've just created a new milestone (_1.0.8_) in order to do that as soon as possible.

PS: The `pip install keyrings.alt` may not be part of the problem, as this is not a pydebrid dependancy and is not linked to guessit/subliminal in any way.

### TL;DR:

**Downgrade guessit by doing `pip install 'guessit==1.0.2' --force-reinstall` as a temporary solution**

**Comment by FrankenLab** [comment]
I realized downgrading was a temporary solution, but I knew you'd have a better solution. I also realized that the keyrings.alt wasn't related to this problem, but it is required to store the login credentials. Pydebrid gave me some error and wouldn't continue without me installing keyrings.alt or some other method of storing credentials.

Thanks for your attention in getting all this fixed. I hope you found some time to enjoy your weekend too!

**Comment by davidderus** [comment]
Be careful, `keyrings.alt` seems to be an alternative implementation for the real pydebrid requirement, `keyring` (https://pypi.python.org/pypi/keyring).

**Comment by FrankenLab** [comment]
This is what's happening with keyrings, I uninstalled keyrings.alt and verified I had keyrings installed:

pydebrid -i
Welcome! It's the first time you launch pydebrid so you need to give us a few credentials.
What's your Alldebrid username? blah
And your password? blahblah
No recommended backend was available. Install the keyrings.alt package if you want to use the non-recommended backends. See README.rst for details.

**Comment by davidderus** [comment]
I found an explanation for that:

```
Python-Keyring is a module that provides a cross-platform interface to store passwords, and has multiple backends. 

Recently that package has been split into two parts: base (python-keyring) which contains only recommended secure backends, and keyrings.alt which contains all the other backends. This ITP is for the latter package.


I am going to make python-keyring recommend keyrings.alt [...]
```

I'm going to add that to the milestone too, under a different issue.
Can you please tell me what distro you use pydebrid on? I've no need for such a requirement on OS X and Ubuntu.

**Comment by FrankenLab** [comment]
Interesting, thanks for sharing that info.  I use pydebrid on FreeBSD, I have a home storage array and I've written a script that collects links I have emailed to me from different forums and feeds them to pydebrid and automatically downloads them for me. It really saves me a lot of time :-)

**Comment by davidderus** [comment]
Normal subliminal (1.1.1) behaviour:

```
Downloading/unpacking subliminal
  Downloading subliminal-1.1.1-py2-none-any.whl (48kB): 48kB downloaded
Downloading/unpacking guessit>=0.9.1,<2.0 (from subliminal)
  Downloading guessit-1.0.3.tar.gz (146kB): 146kB downloaded
```

---

Behaviour when installing pydebrid:

```
Searching for guessit>=0.9.1,<2.0
Reading https://pypi.python.org/simple/guessit/
Best match: guessit 2.0rc8
Downloading https://pypi.python.org/packages/source/g/guessit/guessit-2.0rc8.tar.gz#md5=ce92b95fcd59eb70440c7aac7113bfc5
Processing guessit-2.0rc8.tar.gz
```

---

Solution, updating setup.py:

```
install_requires=['subliminal>=1.1.1,<2', 'guessit<=1.9'...
```

**Comment by davidderus** [comment]
Fixed in 1.0.8 release, now closing the issue.
Feel free to reopen it if needed.
